Package.swift

```diff
@@ -7,7 +7,8 @@ let package = Package(
     name: "Datadog",
     platforms: [
         .iOS(.v11),
-        .tvOS(.v11)
+        .tvOS(.v11),
+        .macOS(.v10_15)
```

Comment:
question/ 10.15 is macOS Catalina, released in 2019 which means supporting 5 major versions of macOS. Was there an extra motivation to reach that far? 

I think  we can save multiple added complexities if targeting macOS Sonoma and Ventura (`.macOS(.v13)`). That includes all the availability checks we added:
```
#available(..., macOS 12.0, *)
#available(..., macOS 10.15, *)
...
```